keras_contrib/applications/densenet.py

```diff
@@ -323,3 +467,146 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
+                                                                    "be a positive number divisible by 4 and greater " \
+                                                                    "than 12"
+
+    # layers in each dense block
+    if type(nb_layers_per_block) is list or type(nb_layers_per_block) is tuple:
+        nb_layers = list(nb_layers_per_block)  # Convert tuple to list
+
+        assert len(nb_layers) == (nb_dense_block + 1), "If list, nb_layer is used as provided. " \
+                                                       "Note that list size must be (nb_dense_block + 1)"
+
+        final_nb_layer = nb_layers[-1]
+        nb_layers = nb_layers[:-1]
+
+    else:
+        final_nb_layer = nb_layers_per_block
+        nb_layers = [nb_layers_per_block] * nb_dense_block
+
+    if bottleneck:
+        nb_layers = [int(layer // 2) for layer in nb_layers]
+
+    # compute initial nb_filter if -1, else accept users initial nb_filter
+    if nb_filter <= 0:
+        nb_filter = 2 * growth_rate
+
+    # compute compression factor
+    compression = 1.0 - reduction
+
+    # Initial convolution
+    x = Convolution2D(48, 3, 3, init="he_uniform", border_mode="same", name="initial_conv2D", bias=False,
+                      W_regularizer=l2(weight_decay))(img_input)
+
+    skip_connection = x
+    skip_list = []
+
+    # Add dense blocks and transition down block
+    for block_idx in range(nb_dense_block):
+        x, nb_filter = __dense_block(x, nb_layers[block_idx], nb_filter, growth_rate, bottleneck=bottleneck,
+                                     dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+        # Skip connection
+        x = merge([x, skip_connection], mode='concat', concat_axis=concat_axis)
+        skip_list.append(x)
+
+        # add transition_block
+        x = __transition_block(x, nb_filter, compression=compression, dropout_rate=dropout_rate,
+                               weight_decay=weight_decay)
+        nb_filter = int(nb_filter * compression)  # this is calculated inside transition_down_block
+
+        # Preserve transition for next skip connection after dense
+        skip_connection = x
+
+    # The last dense_block does not have a transition_down_block
+    x, nb_filter = __dense_block(x, final_nb_layer, nb_filter, growth_rate, bottleneck=bottleneck,
+                                 dropout_rate=dropout_rate, weight_decay=weight_decay)
+
+    if K.image_dim_ordering() == 'th':
+        out_shape = [batchsize, nb_filter, rows // 16, cols // 16]
+    else:
+        out_shape = [batchsize, rows // 16, cols // 16, nb_filter]
+
+    # Add dense blocks and transition up block
+    for block_idx in range(nb_dense_block):
+        if K.image_dim_ordering() != 'th':
+            out_shape[3] = nb_filter
+
+        x = __transition_up_block(x, nb_filters=nb_filter, type=upsampling_type, output_shape=out_shape)
+
+        if K.image_dim_ordering() == 'th':
+            out_shape[2] *= 2
```

Comment:
DenseNet does not use scale parameter at all. 
DenseNet FCN uses 2x scaling because it first performs 2 ^ nb_dense_block downscales and then 2 ^ nb_dense_block upscales. 

All papers I have seen mention 2x scaling to be important. Any larger and it looses too much information when downscaling, and cannot recover it when upscaling.